keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,952 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This document proposes a standard for declaring switch network topology in Kubernetes clusters, representing the hierarchy of nodes, switches, and interconnects. In this context, a switch can refer to a physical network device or a collection of such devices with close proximity and functionality.
+
+## Motivation
+With the rise of multi-node Kubernetes workloads, especially AI workloads that demand intensive inter-node communication, scheduling pods in close network proximity becomes essential.
+
+Some major CSPs already provide a way to discover node network topology.
+Amazon's AWS has implemented [DescribeInstanceTopology API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTopology.html). Google's GCP exposes [Google Cloud SDK](https://cloud.google.com/go/docs/reference/cloud.google.com/go/compute/latest/apiv1) which allows to fetch rack and cluster IDs, from which we could reconstruct network hierarchy. Oracle's OCI provides [Go SDK](https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/gosdk.htm) which allows to get topology related information for their compute nodes.
+
+In addition to CSPs, there is a way to discover switch network topology for some on-prem clusters, but that depends on the supported switch network vendors.
+
+There is an open-source project [topograph](https://github.com/NVIDIA/topograph) that has implemented all the above approaches and has been deployed in production environments.
+
+What is currently missing is a common and standard way to convey the switch network topology information to the Kubernetes environment.
+
+Currently, there is no standardized way to represent this information in Kubernetes, making it challenging to develop control plane components and applications that can leverage network topology awareness.
```

Comment:
OK, but: why does this standardised way need to be part of Kubernetes itself? You need to explain that as well.

A viable alternative could be making something like NVIDIA topograph but vendor neutral, that isn't part of Kubernetes itself.

Reply:
I just updated the "Motivation" part. Hope it will bring more clarity.

Reply:
The end-goal is to use these labels in Kubernetes-native scheduler plugins.
Hence the proposal to make the consistent across CSPs and on-prem.

Reply:
OK, so that needs to go into the KEP text. Without that detail, the motivation isn't obvious.